oci/layout/oci_transport.go

```diff
@@ -64,19 +64,26 @@ type ociReference struct {
 	// If image=="", it means the "only image" in the index.json is used in the case it is a source
 	// for destinations, the image name annotation "image.ref.name" is not added to the index.json
 	image string
+	// Zero-based source manifest index. If sourceIndex==-1, the index will not be valid to point out the source image, only image field will be used.
+	sourceIndex int
 }
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an OCI ImageReference.
 func ParseReference(reference string) (types.ImageReference, error) {
 	dir, image := internal.SplitPathAndImage(reference)
-	return NewReference(dir, image)
+	image, index, err := internal.ParseOCIReferenceName(image)
+	if err != nil {
+		return nil, err
+	}
+	return NewReferenceWithIndex(dir, image, index)
 }
 
-// NewReference returns an OCI reference for a directory and a image.
+// NewReferenceWithIndex returns an OCI reference for a directory and a image， sourceIndex points to an image.
 //
+// If sourceIndex==-1, the index will not be valid to point out the source image, only image will be used.
 // We do not expose an API supplying the resolvedDir; we could, but recomputing it
 // is generally cheap enough that we prefer being confident about the properties of resolvedDir.
-func NewReference(dir, image string) (types.ImageReference, error) {
+func NewReferenceWithIndex(dir, image string, sourceIndex int) (types.ImageReference, error) {
```

Comment:
Probably make this a private `newReference`; let external users call either `NewReference` or `NewIndexReference`.

It would be useful for the two transports to be structurally similar in the API and the implementation approach.